packages/nimble-components/src/select/template.ts

```diff
@@ -58,6 +58,7 @@ SelectOptions
         aria-disabled="${x => x.ariaDisabled}"
         aria-expanded="${x => x.ariaExpanded}"
         aria-haspopup="${x => (x.collapsible ? 'listbox' : null)}"
+        aria-label="${x => x.labelContent}"
```

Comment:
I don't think the select [requires](https://stackoverflow.com/a/56775032/13331204) an explicit `aria-label`, because assistive technology can already infer the label from the slotted content. If that's the case, we should remove the `aria-label`.

I.e. The label and aria-label are both displayed in VoiceOver 

![Screenshot 2024-08-01 at 10 30 25 AM](https://github.com/user-attachments/assets/bee06179-9b6c-45fb-a0d0-8a756c62c178)

Reply:
Without the `aria-label` set on the select, the select still has an accessibility violation about not having an accessible label. I ran a build with that change, [this Storybook build without aria-label](https://60e89457a987cf003efc0a5b-zgzmmrogzr.chromatic.com/?path=/story/components-select--select) shows the violation.
As I noted above, the select and combobox declare their ARIA role in different places (also in the shadow root for the combobox, vs on the template/element itself for the combobox), which may be the reason for the different behavior.

Good catch that the current changes can result in duplicate narration though. Since `aria-label` is set, I probably need to set `aria-hidden` on the `label` element. (I see that in [list-option-group which sets aria-label](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/list-option-group/template.ts#L19) too).

Reply:
Added `aria-hidden` to the `label` element, since `aria-label` is used.